gpii/node_modules/flowManager/src/DefaultSettingsLoader.js

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Default Settings Loader
+ *
+ * Copyright 2018 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = fluid || require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    fs = require("fs"),
+    path = require("path"),
+    JSON5 = require("json5");
+
+fluid.defaults("gpii.defaultSettingsLoader", {
+    gradeNames: ["fluid.component"],
+    // The path to the file that has all QSS supported settings.
+    defaultSettingsInCodeBase: "%gpii-universal/testData/solutions/defaultSettings.json5",
+    members: {
+        gpiiSettingsDir: "@expand:{settingsDir}.getGpiiSettingsDir()"
+        // "defaultSettingsFile" will be set by "onCreate.prepareSettingsFile".
+        // It points to the default settings file in the settings directory. GPII should read from this file.
+        // defaultSettingsFile: null
+    },
+    components: {
+        settingsDir: {
+            type: "gpii.settingsDir"
+        }
+    },
+    listeners: {
+        "onCreate.prepareSettingsFile": {
+            listener: "fluid.set",
```

Comment:
This is too much to do in a listener, as well as being undesirably stateful. Express "defaultSettingsFile" as a member instead